--- a/src/libexec/ldap_alias_manager.pl.in
+++ b/src/libexec/ldap_alias_manager.pl.in
@@ -164,16 +164,8 @@ if ($operation eq 'add') {
         next if ($alias =~ /^\#/);
         next if ($alias =~ /^\s*$/);
 
-        $alias =~ /^([^\s:]+)[\s:]+\"\s*\|\s*(.*)\s+(.*)\s*\"$/;
+        $alias =~ m{^([^\s]+)@[^\s]+[\s]+[^\s]+:.+};
         my $alias_value   = $1;
-        my $command_value = $2;
-        my $maildrop      = $3;
-
-        if ($command_value =~ m/bouncequeue/) {
-            $command_value = $bouncequeue_transport;
-        } else {
-            $command_value = $queue_transport;
-        }
 
         # We substitute all occurrences of + by - for the rest of the
         # attributes, including the dn.
@@ -183,8 +175,6 @@ if ($operation eq 'add') {
 
         my $ldif_dump;
         $data{'list'}{'alias'}    = $alias_value;
-        $data{'list'}{'command'}  = $command_value;
-        $data{'list'}{'maildrop'} = $maildrop;
         unless ($template->parse(\%data, 'ldap_alias_entry.tt2', \$ldif_dump))
         {
             print STDERR "Can't parse ldap_alias_entry.tt2\n";
@@ -242,7 +232,7 @@ elsif ($operation eq 'del') {
             next;
         }
 
-        $alias =~ /^([^\s:]+)[\s:]/;
+        $alias =~ m{^([^\s]+)@[^\s]+[\s]+[^\s]+:.+};
         my $alias_value = $1;
         $alias_value =~ s/\+/\-/g;
 
