--- a/src/cgi/wwsympa.fcgi.in	2018-10-10 00:30:37.636653292 +0200
+++ b/src/cgi/wwsympa.fcgi.in	2018-10-10 12:45:51.168994069 +0200
@@ -422,7 +422,7 @@
     'subindex'   => ['list'],
     'decl_add'  => ['list'],
     'signoff'    => ['list'],
-    'auto_signoff'           => ['list'],
+    'auto_signoff'           => ['list','email'],
     'family_signoff'         => ['family', 'email'],
     'family_signoff_request' => ['family', 'email'],
     #'sigrequest'             => ['list',   'email'],
